### DISCUSSION TITLE
Reading Client's timezone from React Server Components

### DISCUSSION THREAD

**Discussion by adheepgeorgepaul** [vercel/next.js#71247] [Help] [answered]
I need to make a few API calls, and these calls require passing timezone information. I am making these API calls from server components, but if I try to retrieve the timezone from the server, it will return the server's timezone. However, I need to use the client's timezone in the API calls. How can I solve this?

After reading several articles, I found that the x-timezone header can be used to pass timezone information in network requests. However, since the client's timezone can only be detected after the page has loaded in the browser, how can I capture and include the client's timezone in subsequent requests, so that the server can read the header and use it in the API calls?

**Comment by icyJoseph** [comment, marked as answer]
AFAIK, there's no official timezone header, `x-` prefixed headers are sort of unofficial custom if you will.

Perhaps, you have to use a cookie? Set it on the client, upon load, and then look for that cookie when making the server call. You'd probably need some conditional logic for what to do when the cookie is missing and such. Should probably also consider making httponly, which means it'd have to be set by the server, which means you'd need to send the timezone to the server, in a server action or route handler, so that it can be set :)

**Reply by adheepgeorgepaul** [reply]
Thank you so much Icy. I was thinking about using a Cookie yesterday. Your answer cleared up a lot of things. 💯

**Reply by adheepgeorgepaul** [reply]
@icyJoseph Since we are only storing the timezone in the cookie. Is making it httponly necessary? Is timezone a sensitive information?

**Reply by icyJoseph** [reply]
Well it depends on your app, but generally speaking, it should be fine. If someone was to try to spoof this kind of data, would that be a concern for your application? As in, a problem? Likely not, but always double check.

**Reply by adheepgeorgepaul** [reply]
Thanks @icyJoseph
